test/chplcheck/MethodsAfterFields/MethodsAfterFields.prediff

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+while read expectedLine; do
+    echo grep "$expectedLine" MethodsAfterFields.message | echo $? >> MethodsAfterFields.comp.out.tmp
```

Comment:
I don't think this is doing what you want it to do -- at best it's printing `grep theLine messageFile`. It's not actually running grep.

Why do this (i.e. this entire file) at all? 

Could you do:

```Bash
if diff MethodsAfterFields.message MethodsAfterFields.expected; then
    echo "Linter output matched!" >> MethodsAfterFields.comp.out.tmp
else
    echo "Linter output did not match!" >> MethodsAfterFields.comp.out.tmp
fi
```

I suspect this would work because [`diff` returns a nonzero code if the files don't match](https://askubuntu.com/questions/698784/exit-code-of-diff).

Reply:
oops! Thank you for catching this, I forgot to delete the echo. I had originally used diff, but the linter prints the absolute file path, so I can't have one message that is correct for every testing configuration.

Reply:
I suspect that you should be able to strip the (absolute) path to the file by looking at the output of `pwd` and removing it from `$2` to get around this.

Reply:
will do!

Reply:
What are your thoughts on generating the `.good` file by prepending the directory to the partial expected output instead? I think it'd be less of a hassle than stripping the `.comp.out.tmp` file in place in bash

Reply:
Will

```bash
sed -i .tmp "s#$(pwd)/##" $2
```

Not work?

Reply:
Generating `.good` is fine by me as well, but I think it's nicer to see the `.good` file committed with the existing output.

Reply:
> Will
> 
> ```shell
> sed -i .tmp "s#$(pwd)/##" $2
> ```
> 
> Not work to strip out the `pwd`?

Looks like i'll be adding a sed tutorial or two to my todos for the week!